packages/payload/src/collections/operations/updateByID.ts

```diff
@@ -296,7 +296,12 @@ export const updateByIDOperation = async <TSlug extends CollectionSlug>(
     }
 
     if (publishSpecificLocale) {
-      publishedDocWithLocales = await getLatestCollectionVersion({
+      versionSnapshotResult = await beforeChange({
```

Comment:
Can you explain the change of order that we have moving the beforeChange up above the getLatestCollectionVersion.

it seems to me that any beforeChange hooks that return values will be overriden by version data and you wouldn't be able to use the latest version data in your beforeChange? Is that expected? I might not be thinking about this correctly.

Reply:
Yes - I think the confusion here is coming from the double `beforeChange` in this file. What you are saying makes complete sense, the `beforeChange` that runs on the actual doc happens after on line 316:
```ts
    let result = await beforeChange({...})
```

The `versionSnapshotResult = await beforeChange` on line 299 is to save a snapshot that is passed to `saveVersion` on line 367 and is used no where else in the file.
```ts
saveVersion({
        ...
        snapshot: versionSnapshotResult,
      })
```

I only moved the `versionSnapshotResult = await beforeChange` up because when I was re-thinking through the steps that happen here, it felt less confusing to me to take the snapshot first and get it out of the way - but no operational reason it needed to be moved.